<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Lili Wang" />


<title>CACTI peak-window pipeline</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">CACTI peak-window pipeline</h1>
<h4 class="author">Lili Wang</h4>



<div id="introduction" class="section level2">
<h2>1. Introduction</h2>
<p>The <code>cacti</code> package implements the <strong>CACTI
peak-window method</strong> for chromatin QTL mapping.<br />
It is designed to:</p>
<ul>
<li>group nearby peaks into <strong>non-overlapping
windows</strong>,<br />
</li>
<li>residualize peak intensities with respect to covariates,<br />
</li>
<li>test association between genetic variants and <strong>multi-peak
windows</strong> using a principal-component omnibus (PCO) test,
and<br />
</li>
<li>optionally aggregate per-chromosome results and compute
<strong>window-level FDR</strong>.</li>
</ul>
<p>The main user-facing functions in this vignette are:</p>
<ul>
<li><code>cacti_run_chr()</code> – run the full CACTI peak-window
pipeline for <strong>one chromosome</strong>,<br />
</li>
<li><code>cacti_run_genome()</code> – run the pipeline across
<strong>multiple chromosomes</strong> and add FDR.</li>
</ul>
<p>This vignette explains:</p>
<ol style="list-style-type: decimal">
<li>The overall workflow.<br />
</li>
<li>The expected <strong>input file formats</strong>.<br />
</li>
<li>The <strong>output files</strong> produced by the pipeline.<br />
</li>
<li>Example usage with bundled toy data under
<code>inst/extdata/</code>.</li>
</ol>
<hr />
</div>
<div id="overview-of-the-peak-window-pipeline" class="section level2">
<h2>2. Overview of the peak-window pipeline</h2>
<p>The CACTI peak-window pipeline has three main steps:</p>
<ol style="list-style-type: decimal">
<li><strong>Group peaks into non-overlapping windows</strong>
<ul>
<li>Using genomic coordinates (chromosome, start, end) of peaks,<br />
</li>
<li>Divide the genome into fixed-size windows (e.g. 50 kb),<br />
</li>
<li>Assign peaks to these windows and create a <strong>window (group)
table</strong>.</li>
</ul></li>
<li><strong>Residualize phenotypes with respect to covariates</strong>
<ul>
<li>Input: peak intensity matrix (features x samples) and covariate
matrix (covariates x samples),<br />
</li>
<li>Fit linear models <code>peak ~ covariates</code>,<br />
</li>
<li>Save residualized intensity values in the <strong>same
format</strong> as the original matrix, as one of the inputs for the
association test.</li>
</ul></li>
<li><strong>Run association tests on peak windows</strong>
<ul>
<li>Input: cis-QTL summary statistics <code>z</code> for peak–SNP pairs,
peak windows, and residualized phenotype,<br />
</li>
<li>For each window:
<ul>
<li>construct a Z-score matrix (variants x peaks),</li>
<li>compute a PCO-based multi-peak p-value if the number of peaks ≥
<code>min_peaks</code>,</li>
<li>otherwise fall back to a univariate p-value,</li>
</ul></li>
<li>Output: one p-value file per chromosome, one row per
<code>(group, snp)</code> pair.</li>
</ul></li>
</ol>
<p>Across all chromosomes, <code>cacti_add_fdr()</code> (used by
<code>cacti_run_genome()</code>) aggregates p-values and computes
<strong>q-values</strong> for the top hit per window.</p>
<hr />
</div>
<div id="input-file-formats" class="section level2">
<h2>3. Input file formats</h2>
<p>This section describes the expected columns and layout of each input
file type.</p>
<p>We will also inspect the <strong>toy dataset</strong> bundled with
the package in:</p>
<pre class="text"><code>inst/extdata/
  test_pheno_meta.bed
  test_pheno.txt
  test_covariates.txt
  test_qtl_sum_stats_chr5.txt.gz</code></pre>
<div id="peak-metadata-pheno-meta-bed-like" class="section level3">
<h3>3.1 Peak metadata (pheno meta / BED-like)</h3>
<p>The <strong>peak metadata</strong> file describes the genomic
coordinates and IDs of peaks.</p>
<ul>
<li>Used by: <code>cacti_group_peak_window()</code>,
<code>cacti_run_chr()</code>, <code>cacti_run_genome()</code></li>
<li>Argument: <code>file_pheno_meta</code></li>
<li>Example file: <code>test_pheno_meta.bed</code></li>
</ul>
<p>Required columns:</p>
<ul>
<li><code>phe_chr</code> – chromosome name, typically with
<code>&quot;chr&quot;</code> prefix (e.g. <code>&quot;chr5&quot;</code>),<br />
</li>
<li><code>phe_from</code> – peak start position (integer; must be
<code>&lt; phe_to</code>),<br />
</li>
<li><code>phe_to</code> – peak end position (integer),<br />
</li>
<li><code>phe_id</code> – unique peak identifier.</li>
</ul>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="fu">library</span>(cacti)</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a>file_pheno_meta <span class="ot">&lt;-</span> <span class="fu">system.file</span>(</span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a>  <span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;test_pheno_meta.bed&quot;</span>,</span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a>  <span class="at">package =</span> <span class="st">&quot;cacti&quot;</span></span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a>)</span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a></span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a><span class="fu">head</span>(data.table<span class="sc">::</span><span class="fu">fread</span>(file_pheno_meta))</span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a><span class="co">#&gt;                phe_id phe_chr phe_from phe_to</span></span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a><span class="co">#&gt;                &lt;char&gt;  &lt;char&gt;    &lt;int&gt;  &lt;int&gt;</span></span>
<span id="cb2-11"><a href="#cb2-11" tabindex="-1"></a><span class="co">#&gt; 1:   chr5_11429_11873    chr5    11429  11873</span></span>
<span id="cb2-12"><a href="#cb2-12" tabindex="-1"></a><span class="co">#&gt; 2: chr5_217285_220962    chr5   217285 220962</span></span>
<span id="cb2-13"><a href="#cb2-13" tabindex="-1"></a><span class="co">#&gt; 3: chr5_223012_223844    chr5   223012 223844</span></span>
<span id="cb2-14"><a href="#cb2-14" tabindex="-1"></a><span class="co">#&gt; 4: chr5_270610_273325    chr5   270610 273325</span></span>
<span id="cb2-15"><a href="#cb2-15" tabindex="-1"></a><span class="co">#&gt; 5: chr5_320783_323201    chr5   320783 323201</span></span>
<span id="cb2-16"><a href="#cb2-16" tabindex="-1"></a><span class="co">#&gt; 6: chr5_343419_345060    chr5   343419 345060</span></span></code></pre></div>
<hr />
</div>
<div id="phenotype-matrix-peak-intensity" class="section level3">
<h3>3.2 Phenotype matrix (peak intensity)</h3>
<p>The <strong>phenotype matrix</strong> contains peak intensities (or
other quantitative trait values).</p>
<ul>
<li>Used by: <code>cacti_pheno_cov_residual()</code>,
<code>cacti_run_chr()</code>, <code>cacti_run_genome()</code></li>
<li>Argument: <code>file_pheno</code></li>
<li>Example file: <code>test_pheno.txt</code></li>
</ul>
<p>Layout:</p>
<ul>
<li>Rows = features (peaks).<br />
</li>
<li>First column = feature ID (matching <code>phe_id</code> in
metadata).<br />
</li>
<li>Remaining columns = samples.</li>
</ul>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a>file_pheno <span class="ot">&lt;-</span> <span class="fu">system.file</span>(</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a>  <span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;test_pheno.txt&quot;</span>,</span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a>  <span class="at">package =</span> <span class="st">&quot;cacti&quot;</span></span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a>)</span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a></span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a><span class="fu">head</span>(data.table<span class="sc">::</span><span class="fu">fread</span>(file_pheno))[, <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a><span class="co">#&gt;                        ID       AF04      AF06       AF08      AF12</span></span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a><span class="co">#&gt;                    &lt;char&gt;      &lt;num&gt;     &lt;num&gt;      &lt;num&gt;     &lt;num&gt;</span></span>
<span id="cb3-9"><a href="#cb3-9" tabindex="-1"></a><span class="co">#&gt; 1: chr5_10248638_10251938  6.1639272 6.5171965  6.0122366 6.0046710</span></span>
<span id="cb3-10"><a href="#cb3-10" tabindex="-1"></a><span class="co">#&gt; 2: chr5_10264481_10265888  0.2981699 0.2977740  2.0098907 0.4789852</span></span>
<span id="cb3-11"><a href="#cb3-11" tabindex="-1"></a><span class="co">#&gt; 3: chr5_10265755_10268539  1.6214124 2.1306168  3.1048357 0.7830301</span></span>
<span id="cb3-12"><a href="#cb3-12" tabindex="-1"></a><span class="co">#&gt; 4: chr5_10307340_10308285  0.7222144 0.8835924  0.2871122 0.5861384</span></span>
<span id="cb3-13"><a href="#cb3-13" tabindex="-1"></a><span class="co">#&gt; 5: chr5_10311409_10312352  0.0350794 1.3397306 -0.2311209 0.3488682</span></span>
<span id="cb3-14"><a href="#cb3-14" tabindex="-1"></a><span class="co">#&gt; 6: chr5_10333519_10334110 -0.1660226 0.8808080  0.5575939 1.1806155</span></span></code></pre></div>
<hr />
</div>
<div id="covariate-matrix" class="section level3">
<h3>3.3 Covariate matrix</h3>
<p>The <strong>covariate matrix</strong> encodes covariates for each
sample (e.g. PCs, batch, sex, etc.).</p>
<ul>
<li>Used by: <code>cacti_pheno_cov_residual()</code>,
<code>cacti_run_chr()</code>, <code>cacti_run_genome()</code></li>
<li>Argument: <code>file_cov</code></li>
<li>Example file: <code>test_covariates.txt</code></li>
</ul>
<p>Layout:</p>
<ul>
<li>Rows = covariates.<br />
</li>
<li>First column = covariate ID.<br />
</li>
<li>Remaining columns = samples.</li>
</ul>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a>file_cov <span class="ot">&lt;-</span> <span class="fu">system.file</span>(</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a>  <span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;test_covariates.txt&quot;</span>,</span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a>  <span class="at">package =</span> <span class="st">&quot;cacti&quot;</span></span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a>)</span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a></span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a><span class="fu">head</span>(data.table<span class="sc">::</span><span class="fu">fread</span>(file_cov))[, <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a><span class="co">#&gt;        ID       AF04       AF06       AF08       AF12</span></span>
<span id="cb4-8"><a href="#cb4-8" tabindex="-1"></a><span class="co">#&gt;    &lt;char&gt;      &lt;num&gt;      &lt;num&gt;      &lt;num&gt;      &lt;num&gt;</span></span>
<span id="cb4-9"><a href="#cb4-9" tabindex="-1"></a><span class="co">#&gt; 1:   cov1 -0.1696051 -0.2304933 -0.2132352 -0.2220646</span></span></code></pre></div>
<hr />
</div>
<div id="qtl-summary-statistics" class="section level3">
<h3>3.4 QTL summary statistics</h3>
<p>The <strong>cis-QTL summary stats</strong> file contains association
summary statistics for peak–SNP pairs.</p>
<ul>
<li>Used by: <code>cacti_cal_p()</code>, <code>cacti_run_chr()</code>,
<code>cacti_run_genome()</code></li>
<li>Argument: <code>qtl_file</code> (per chromosome) or
<code>qtl_files</code> (multiple chromosomes, given as a
vector/template)</li>
<li>Example file: <code>test_qtl_sum_stats_chr5.txt.gz</code></li>
</ul>
<p>Required columns:</p>
<ul>
<li><code>phe_id</code> – peak ID (matching <code>phe_id</code> in
metadata and <code>ID</code> in phenotype matrix),<br />
</li>
<li><code>var_id</code> – variant ID,<br />
</li>
<li><code>z</code> – z-score for the association between that variant
and that peak.</li>
</ul>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a>qtl_file <span class="ot">&lt;-</span> <span class="fu">system.file</span>(</span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a>  <span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;test_qtl_sum_stats_chr5.txt.gz&quot;</span>,</span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a>  <span class="at">package =</span> <span class="st">&quot;cacti&quot;</span></span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a>)</span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a></span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a><span class="fu">head</span>(data.table<span class="sc">::</span><span class="fu">fread</span>(qtl_file))</span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a><span class="co">#&gt;              phe_id      var_id         z</span></span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a><span class="co">#&gt;              &lt;char&gt;      &lt;char&gt;     &lt;num&gt;</span></span>
<span id="cb5-9"><a href="#cb5-9" tabindex="-1"></a><span class="co">#&gt; 1: chr5_11429_11873 5_12041_T_A 0.2715305</span></span>
<span id="cb5-10"><a href="#cb5-10" tabindex="-1"></a><span class="co">#&gt; 2: chr5_11429_11873 5_12114_G_C 0.1525914</span></span>
<span id="cb5-11"><a href="#cb5-11" tabindex="-1"></a><span class="co">#&gt; 3: chr5_11429_11873 5_12225_A_G 1.0831622</span></span>
<span id="cb5-12"><a href="#cb5-12" tabindex="-1"></a><span class="co">#&gt; 4: chr5_11429_11873 5_13018_A_C 0.9401877</span></span>
<span id="cb5-13"><a href="#cb5-13" tabindex="-1"></a><span class="co">#&gt; 5: chr5_11429_11873 5_13114_C_G 0.9401877</span></span>
<span id="cb5-14"><a href="#cb5-14" tabindex="-1"></a><span class="co">#&gt; 6: chr5_11429_11873 5_13125_T_G 0.9401877</span></span></code></pre></div>
<hr />
</div>
</div>
<div id="output-files-from-cacti_run_chr" class="section level2">
<h2>4. Output files from <code>cacti_run_chr()</code></h2>
<p><code>cacti_run_chr()</code> is the main function for running the
CACTI peak-window pipeline on a <strong>single chromosome</strong>.</p>
<div id="overview" class="section level3">
<h3>4.1 Overview</h3>
<p>Calling:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a>res_single_chr <span class="ot">&lt;-</span> <span class="fu">cacti_run_chr</span>(</span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a>  <span class="at">window_size     =</span> <span class="st">&quot;50kb&quot;</span>,</span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a>  <span class="at">file_pheno_meta =</span> <span class="sc">&lt;</span>peak metadata<span class="sc">&gt;</span>,</span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a>  <span class="at">file_pheno      =</span> <span class="sc">&lt;</span>phenotype matrix<span class="sc">&gt;</span>,</span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a>  <span class="at">file_cov        =</span> <span class="sc">&lt;</span>covariate matrix<span class="sc">&gt;</span>,</span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a>  <span class="at">chr             =</span> <span class="st">&quot;chr5&quot;</span>,</span>
<span id="cb6-7"><a href="#cb6-7" tabindex="-1"></a>  <span class="at">qtl_file        =</span> <span class="sc">&lt;</span>QTL summary stats <span class="cf">for</span> chr5<span class="sc">&gt;</span>,</span>
<span id="cb6-8"><a href="#cb6-8" tabindex="-1"></a>  <span class="at">out_prefix      =</span> <span class="sc">&lt;</span>output prefix<span class="sc">&gt;</span>,</span>
<span id="cb6-9"><a href="#cb6-9" tabindex="-1"></a>  <span class="at">min_peaks       =</span> <span class="dv">2</span></span>
<span id="cb6-10"><a href="#cb6-10" tabindex="-1"></a>)</span></code></pre></div>
<p>will:</p>
<ol style="list-style-type: decimal">
<li>Construct peak windows on all chromosomes (using
<code>window_size</code>).<br />
</li>
<li>Residualize the phenotype matrix.<br />
</li>
<li>Run the CACTI association test on the specified chromosome
(<code>chr</code>).</li>
</ol>
<p>It returns a named list of output paths, and writes the following
files:</p>
</div>
<div id="window-level-grouping-file" class="section level3">
<h3>4.2 Window-level grouping file</h3>
<ul>
<li>Path: <code>res_single_chr$file_peak_group</code><br />
</li>
<li>Produced by: <code>cacti_group_peak_window()</code></li>
</ul>
<p>Columns:</p>
<ul>
<li><code>group</code> – window (group) ID (e.g. <code>&quot;G1&quot;</code>,
<code>&quot;G2&quot;</code>, …),<br />
</li>
<li><code>group_chr</code> – chromosome for the window,<br />
</li>
<li><code>group_from</code> – window start position,<br />
</li>
<li><code>group_to</code> – window end position,<br />
</li>
<li><code>pid_group</code> – semicolon-separated list of peak IDs in the
window,<br />
</li>
<li><code>n_pid_group</code> – number of peaks in the window.</li>
</ul>
</div>
<div id="peak-level-grouping-file" class="section level3">
<h3>4.3 Peak-level grouping file</h3>
<ul>
<li>Path: <code>res_single_chr$file_peak_group_peaklevel</code></li>
</ul>
<p>Columns:</p>
<ul>
<li><code>phe_id</code> – peak ID,<br />
</li>
<li><code>phe_chr</code> – chromosome,<br />
</li>
<li><code>phe_from</code>, <code>phe_to</code> – peak start/end,<br />
</li>
<li><code>group</code> – window ID to which the peak belongs,<br />
</li>
<li><code>n_pid_group</code> – number of peaks in that window.</li>
</ul>
</div>
<div id="residualized-phenotype-matrix" class="section level3">
<h3>4.4 Residualized phenotype matrix</h3>
<ul>
<li>Path: <code>res_single_chr$file_pheno_cov_residual</code></li>
</ul>
<p>Same layout as the input phenotype matrix:</p>
<ul>
<li>Rows = features (peak IDs),<br />
</li>
<li>First column = <code>ID</code> (peak IDs),<br />
</li>
<li>Remaining columns = residualized values for each sample.</li>
</ul>
</div>
<div id="per-window-p-value-file-for-the-chromosome" class="section level3">
<h3>4.5 Per-window p-value file for the chromosome</h3>
<ul>
<li>Path: <code>res_single_chr$file_p_peak_group</code></li>
</ul>
<p>Each row corresponds to a <code>(group, snp)</code> pair:</p>
<ul>
<li><code>group</code> – window ID,<br />
</li>
<li><code>snp</code> – variant ID,<br />
</li>
<li><code>pval</code> – window-level p-value for that variant.
<ul>
<li>For windows with ≥ <code>min_peaks</code> peaks, this is the
PCO-based multi-peak p-value.<br />
</li>
<li>For single-peak windows, it is a univariate p-value derived from
<code>z</code>.</li>
</ul></li>
</ul>
<hr />
</div>
</div>
<div id="per-chromosome-example-cacti_run_chr" class="section level2">
<h2>5. Per-chromosome example: <code>cacti_run_chr()</code></h2>
<p>Below is an example of running the full pipeline on the <strong>toy
chr5 data</strong> included in <code>inst/extdata/</code>.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="fu">library</span>(cacti)</span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a>file_pheno_meta <span class="ot">&lt;-</span> <span class="fu">system.file</span>(</span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a>  <span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;test_pheno_meta.bed&quot;</span>,</span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a>  <span class="at">package =</span> <span class="st">&quot;cacti&quot;</span></span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a>)</span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a></span>
<span id="cb7-8"><a href="#cb7-8" tabindex="-1"></a>file_pheno <span class="ot">&lt;-</span> <span class="fu">system.file</span>(</span>
<span id="cb7-9"><a href="#cb7-9" tabindex="-1"></a>  <span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;test_pheno.txt&quot;</span>,</span>
<span id="cb7-10"><a href="#cb7-10" tabindex="-1"></a>  <span class="at">package =</span> <span class="st">&quot;cacti&quot;</span></span>
<span id="cb7-11"><a href="#cb7-11" tabindex="-1"></a>)</span>
<span id="cb7-12"><a href="#cb7-12" tabindex="-1"></a></span>
<span id="cb7-13"><a href="#cb7-13" tabindex="-1"></a>file_cov <span class="ot">&lt;-</span> <span class="fu">system.file</span>(</span>
<span id="cb7-14"><a href="#cb7-14" tabindex="-1"></a>  <span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;test_covariates.txt&quot;</span>,</span>
<span id="cb7-15"><a href="#cb7-15" tabindex="-1"></a>  <span class="at">package =</span> <span class="st">&quot;cacti&quot;</span></span>
<span id="cb7-16"><a href="#cb7-16" tabindex="-1"></a>)</span>
<span id="cb7-17"><a href="#cb7-17" tabindex="-1"></a></span>
<span id="cb7-18"><a href="#cb7-18" tabindex="-1"></a>qtl_file <span class="ot">&lt;-</span> <span class="fu">system.file</span>(</span>
<span id="cb7-19"><a href="#cb7-19" tabindex="-1"></a>  <span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;test_qtl_sum_stats_chr5.txt.gz&quot;</span>,</span>
<span id="cb7-20"><a href="#cb7-20" tabindex="-1"></a>  <span class="at">package =</span> <span class="st">&quot;cacti&quot;</span></span>
<span id="cb7-21"><a href="#cb7-21" tabindex="-1"></a>)</span>
<span id="cb7-22"><a href="#cb7-22" tabindex="-1"></a></span>
<span id="cb7-23"><a href="#cb7-23" tabindex="-1"></a><span class="co"># Use a temporary prefix to avoid cluttering the working directory</span></span>
<span id="cb7-24"><a href="#cb7-24" tabindex="-1"></a>out_prefix <span class="ot">&lt;-</span> <span class="fu">tempfile</span>(<span class="st">&quot;cacti_chr5_&quot;</span>)</span>
<span id="cb7-25"><a href="#cb7-25" tabindex="-1"></a></span>
<span id="cb7-26"><a href="#cb7-26" tabindex="-1"></a>res_single_chr <span class="ot">&lt;-</span> <span class="fu">cacti_run_chr</span>(</span>
<span id="cb7-27"><a href="#cb7-27" tabindex="-1"></a>  <span class="at">window_size     =</span> <span class="st">&quot;50kb&quot;</span>,</span>
<span id="cb7-28"><a href="#cb7-28" tabindex="-1"></a>  <span class="at">file_pheno_meta =</span> file_pheno_meta,</span>
<span id="cb7-29"><a href="#cb7-29" tabindex="-1"></a>  <span class="at">file_pheno      =</span> file_pheno,</span>
<span id="cb7-30"><a href="#cb7-30" tabindex="-1"></a>  <span class="at">file_cov        =</span> file_cov,</span>
<span id="cb7-31"><a href="#cb7-31" tabindex="-1"></a>  <span class="at">chr             =</span> <span class="st">&quot;chr5&quot;</span>,</span>
<span id="cb7-32"><a href="#cb7-32" tabindex="-1"></a>  <span class="at">qtl_file        =</span> qtl_file,</span>
<span id="cb7-33"><a href="#cb7-33" tabindex="-1"></a>  <span class="at">out_prefix      =</span> out_prefix,</span>
<span id="cb7-34"><a href="#cb7-34" tabindex="-1"></a>  <span class="at">min_peaks       =</span> <span class="dv">2</span></span>
<span id="cb7-35"><a href="#cb7-35" tabindex="-1"></a>)</span>
<span id="cb7-36"><a href="#cb7-36" tabindex="-1"></a><span class="co">#&gt; [1/3] Group peaks into non-overlapping windows …</span></span>
<span id="cb7-37"><a href="#cb7-37" tabindex="-1"></a><span class="co">#&gt; [2/3] Residualize phenotype by covariates …</span></span>
<span id="cb7-38"><a href="#cb7-38" tabindex="-1"></a><span class="co">#&gt; [3/3] Run pvalue for chr5 …</span></span>
<span id="cb7-39"><a href="#cb7-39" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-40"><a href="#cb7-40" tabindex="-1"></a><span class="co">#&gt;  CACTI peak-window pipeline for chr5 completed successfully!</span></span>
<span id="cb7-41"><a href="#cb7-41" tabindex="-1"></a><span class="co">#&gt; Run summary:</span></span>
<span id="cb7-42"><a href="#cb7-42" tabindex="-1"></a><span class="co">#&gt; Chromosome:chr5</span></span>
<span id="cb7-43"><a href="#cb7-43" tabindex="-1"></a><span class="co">#&gt; Window size:50kb</span></span>
<span id="cb7-44"><a href="#cb7-44" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-45"><a href="#cb7-45" tabindex="-1"></a><span class="co">#&gt; Input files:</span></span>
<span id="cb7-46"><a href="#cb7-46" tabindex="-1"></a><span class="co">#&gt; Peaks BED:/tmp/RtmppxoEWG/temp_libpath311827286f11e/cacti/extdata/test_pheno_meta.bed</span></span>
<span id="cb7-47"><a href="#cb7-47" tabindex="-1"></a><span class="co">#&gt; Phenotype:/tmp/RtmppxoEWG/temp_libpath311827286f11e/cacti/extdata/test_pheno.txt</span></span>
<span id="cb7-48"><a href="#cb7-48" tabindex="-1"></a><span class="co">#&gt; Covariate:/tmp/RtmppxoEWG/temp_libpath311827286f11e/cacti/extdata/test_covariates.txt</span></span>
<span id="cb7-49"><a href="#cb7-49" tabindex="-1"></a><span class="co">#&gt; QTL summary stats:/tmp/RtmppxoEWG/temp_libpath311827286f11e/cacti/extdata/test_qtl_sum_stats_chr5.txt.gz</span></span>
<span id="cb7-50"><a href="#cb7-50" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-51"><a href="#cb7-51" tabindex="-1"></a><span class="co">#&gt; Output files:</span></span>
<span id="cb7-52"><a href="#cb7-52" tabindex="-1"></a><span class="co">#&gt; Grouped peak: /tmp/RtmpgYDXR8/cacti_chr5_3137f7292b8a_peak_group_window50kb.txt</span></span>
<span id="cb7-53"><a href="#cb7-53" tabindex="-1"></a><span class="co">#&gt; Grouped peak (peak as row):/tmp/RtmpgYDXR8/cacti_chr5_3137f7292b8a_peak_group_window50kb_peak_as_row.txt</span></span>
<span id="cb7-54"><a href="#cb7-54" tabindex="-1"></a><span class="co">#&gt; Residual phenotype:/tmp/RtmpgYDXR8/cacti_chr5_3137f7292b8a_pheno_cov_residual.txt</span></span>
<span id="cb7-55"><a href="#cb7-55" tabindex="-1"></a><span class="co">#&gt; Pval results (chr5): /tmp/RtmpgYDXR8/cacti_chr5_3137f7292b8a_pval_window50kb_chr5.txt.gz</span></span>
<span id="cb7-56"><a href="#cb7-56" tabindex="-1"></a></span>
<span id="cb7-57"><a href="#cb7-57" tabindex="-1"></a>res_single_chr</span>
<span id="cb7-58"><a href="#cb7-58" tabindex="-1"></a><span class="co">#&gt; $file_peak_group</span></span>
<span id="cb7-59"><a href="#cb7-59" tabindex="-1"></a><span class="co">#&gt; [1] &quot;/tmp/RtmpgYDXR8/cacti_chr5_3137f7292b8a_peak_group_window50kb.txt&quot;</span></span>
<span id="cb7-60"><a href="#cb7-60" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-61"><a href="#cb7-61" tabindex="-1"></a><span class="co">#&gt; $file_peak_group_peaklevel</span></span>
<span id="cb7-62"><a href="#cb7-62" tabindex="-1"></a><span class="co">#&gt; [1] &quot;/tmp/RtmpgYDXR8/cacti_chr5_3137f7292b8a_peak_group_window50kb_peak_as_row.txt&quot;</span></span>
<span id="cb7-63"><a href="#cb7-63" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-64"><a href="#cb7-64" tabindex="-1"></a><span class="co">#&gt; $file_pheno_cov_residual</span></span>
<span id="cb7-65"><a href="#cb7-65" tabindex="-1"></a><span class="co">#&gt; [1] &quot;/tmp/RtmpgYDXR8/cacti_chr5_3137f7292b8a_pheno_cov_residual.txt&quot;</span></span>
<span id="cb7-66"><a href="#cb7-66" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb7-67"><a href="#cb7-67" tabindex="-1"></a><span class="co">#&gt; $file_p_peak_group</span></span>
<span id="cb7-68"><a href="#cb7-68" tabindex="-1"></a><span class="co">#&gt; [1] &quot;/tmp/RtmpgYDXR8/cacti_chr5_3137f7292b8a_pval_window50kb_chr5.txt.gz&quot;</span></span></code></pre></div>
<hr />
</div>
<div id="genome-wide-wrapper-cacti_run_genome" class="section level2">
<h2>6. Genome-wide wrapper: <code>cacti_run_genome()</code></h2>
<p><code>cacti_run_genome()</code> provides a convenience wrapper for
running the pipeline across <strong>multiple chromosomes</strong>, then
aggregating the p-values and adding FDR via
<code>cacti_add_fdr()</code>.</p>
<p>Conceptually, it:</p>
<ol style="list-style-type: decimal">
<li>Loops over <code>chrs</code> and calls <code>cacti_run_chr()</code>
for each one.<br />
</li>
<li>Collects all per-chromosome p-value files.<br />
</li>
<li>Calls <code>cacti_add_fdr()</code> once to compute window-level
q-values across all windows and chromosomes.</li>
</ol>
<p>The simplest usage is when you have one QTL file per chromosome and
pass either:</p>
<ul>
<li>a character vector <code>qtl_files</code> with one path per
chromosome, or<br />
</li>
<li>a template with <code>&quot;{chr}&quot;</code> placeholder,
e.g. <code>&quot;extdata/test_qtl_sum_stats_{chr}.txt.gz&quot;</code>.</li>
</ul>
<p>In the toy example, we only use data for <strong>chr5</strong>, but
this still demonstrates how to call the function.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a>file_pheno_meta <span class="ot">&lt;-</span> <span class="fu">system.file</span>(</span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a>  <span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;test_pheno_meta.bed&quot;</span>,</span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a>  <span class="at">package =</span> <span class="st">&quot;cacti&quot;</span></span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a>)</span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a></span>
<span id="cb8-6"><a href="#cb8-6" tabindex="-1"></a>file_pheno <span class="ot">&lt;-</span> <span class="fu">system.file</span>(</span>
<span id="cb8-7"><a href="#cb8-7" tabindex="-1"></a>  <span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;test_pheno.txt&quot;</span>,</span>
<span id="cb8-8"><a href="#cb8-8" tabindex="-1"></a>  <span class="at">package =</span> <span class="st">&quot;cacti&quot;</span></span>
<span id="cb8-9"><a href="#cb8-9" tabindex="-1"></a>)</span>
<span id="cb8-10"><a href="#cb8-10" tabindex="-1"></a></span>
<span id="cb8-11"><a href="#cb8-11" tabindex="-1"></a>file_cov <span class="ot">&lt;-</span> <span class="fu">system.file</span>(</span>
<span id="cb8-12"><a href="#cb8-12" tabindex="-1"></a>  <span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;test_covariates.txt&quot;</span>,</span>
<span id="cb8-13"><a href="#cb8-13" tabindex="-1"></a>  <span class="at">package =</span> <span class="st">&quot;cacti&quot;</span></span>
<span id="cb8-14"><a href="#cb8-14" tabindex="-1"></a>)</span>
<span id="cb8-15"><a href="#cb8-15" tabindex="-1"></a></span>
<span id="cb8-16"><a href="#cb8-16" tabindex="-1"></a>qtl_file <span class="ot">&lt;-</span> <span class="fu">system.file</span>(</span>
<span id="cb8-17"><a href="#cb8-17" tabindex="-1"></a>  <span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;test_qtl_sum_stats_chr5.txt.gz&quot;</span>,</span>
<span id="cb8-18"><a href="#cb8-18" tabindex="-1"></a>  <span class="at">package =</span> <span class="st">&quot;cacti&quot;</span></span>
<span id="cb8-19"><a href="#cb8-19" tabindex="-1"></a>)</span>
<span id="cb8-20"><a href="#cb8-20" tabindex="-1"></a></span>
<span id="cb8-21"><a href="#cb8-21" tabindex="-1"></a>out_prefix <span class="ot">&lt;-</span> <span class="fu">tempfile</span>(<span class="st">&quot;cacti_genome_&quot;</span>)</span>
<span id="cb8-22"><a href="#cb8-22" tabindex="-1"></a></span>
<span id="cb8-23"><a href="#cb8-23" tabindex="-1"></a>res_genome <span class="ot">&lt;-</span> <span class="fu">cacti_run_genome</span>(</span>
<span id="cb8-24"><a href="#cb8-24" tabindex="-1"></a>  <span class="at">window_size     =</span> <span class="st">&quot;50kb&quot;</span>,</span>
<span id="cb8-25"><a href="#cb8-25" tabindex="-1"></a>  <span class="at">file_pheno_meta =</span> file_pheno_meta,</span>
<span id="cb8-26"><a href="#cb8-26" tabindex="-1"></a>  <span class="at">file_pheno      =</span> file_pheno,</span>
<span id="cb8-27"><a href="#cb8-27" tabindex="-1"></a>  <span class="at">file_cov        =</span> file_cov,</span>
<span id="cb8-28"><a href="#cb8-28" tabindex="-1"></a>  <span class="at">chrs            =</span> <span class="st">&quot;chr5&quot;</span>,                 <span class="co"># toy example: one chromosome</span></span>
<span id="cb8-29"><a href="#cb8-29" tabindex="-1"></a>  <span class="at">qtl_files       =</span> qtl_file,</span>
<span id="cb8-30"><a href="#cb8-30" tabindex="-1"></a>  <span class="at">out_prefix      =</span> out_prefix,</span>
<span id="cb8-31"><a href="#cb8-31" tabindex="-1"></a>  <span class="at">file_fdr_out    =</span> <span class="fu">file.path</span>(<span class="fu">tempdir</span>(), <span class="st">&quot;cacti_fdr_chr5.txt.gz&quot;</span>)</span>
<span id="cb8-32"><a href="#cb8-32" tabindex="-1"></a>)</span>
<span id="cb8-33"><a href="#cb8-33" tabindex="-1"></a><span class="co">#&gt; === [I/III] Running CACTI peak-window pipeline genome-wide ===</span></span>
<span id="cb8-34"><a href="#cb8-34" tabindex="-1"></a><span class="co">#&gt; [1/3] Group peaks into non-overlapping windows …</span></span>
<span id="cb8-35"><a href="#cb8-35" tabindex="-1"></a><span class="co">#&gt; [2/3] Residualize phenotype by covariates …</span></span>
<span id="cb8-36"><a href="#cb8-36" tabindex="-1"></a><span class="co">#&gt; [3/3] Run pvalue for chr5 …</span></span>
<span id="cb8-37"><a href="#cb8-37" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-38"><a href="#cb8-38" tabindex="-1"></a><span class="co">#&gt;  CACTI peak-window pipeline for chr5 completed successfully!</span></span>
<span id="cb8-39"><a href="#cb8-39" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-40"><a href="#cb8-40" tabindex="-1"></a><span class="co">#&gt; Input files:</span></span>
<span id="cb8-41"><a href="#cb8-41" tabindex="-1"></a><span class="co">#&gt; Peaks BED:/tmp/RtmppxoEWG/temp_libpath311827286f11e/cacti/extdata/test_pheno_meta.bed</span></span>
<span id="cb8-42"><a href="#cb8-42" tabindex="-1"></a><span class="co">#&gt; Phenotype:/tmp/RtmppxoEWG/temp_libpath311827286f11e/cacti/extdata/test_pheno.txt</span></span>
<span id="cb8-43"><a href="#cb8-43" tabindex="-1"></a><span class="co">#&gt; Covariate:/tmp/RtmppxoEWG/temp_libpath311827286f11e/cacti/extdata/test_covariates.txt</span></span>
<span id="cb8-44"><a href="#cb8-44" tabindex="-1"></a><span class="co">#&gt; QTL summary stats:/tmp/RtmppxoEWG/temp_libpath311827286f11e/cacti/extdata/test_qtl_sum_stats_chr5.txt.gz</span></span>
<span id="cb8-45"><a href="#cb8-45" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-46"><a href="#cb8-46" tabindex="-1"></a><span class="co">#&gt; Output files:</span></span>
<span id="cb8-47"><a href="#cb8-47" tabindex="-1"></a><span class="co">#&gt; Grouped peak: /tmp/RtmpgYDXR8/cacti_genome_3137f182f5581_peak_group_window50kb.txt</span></span>
<span id="cb8-48"><a href="#cb8-48" tabindex="-1"></a><span class="co">#&gt; Grouped peak (peak as row):/tmp/RtmpgYDXR8/cacti_genome_3137f182f5581_peak_group_window50kb_peak_as_row.txt</span></span>
<span id="cb8-49"><a href="#cb8-49" tabindex="-1"></a><span class="co">#&gt; Residual phenotype:/tmp/RtmpgYDXR8/cacti_genome_3137f182f5581_pheno_cov_residual.txt</span></span>
<span id="cb8-50"><a href="#cb8-50" tabindex="-1"></a><span class="co">#&gt; Pval results (chr5): /tmp/RtmpgYDXR8/cacti_genome_3137f182f5581_pval_window50kb_chr5.txt.gz</span></span>
<span id="cb8-51"><a href="#cb8-51" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-52"><a href="#cb8-52" tabindex="-1"></a><span class="co">#&gt; === [II/III] Adding FDR across windows and chromosomes ===</span></span>
<span id="cb8-53"><a href="#cb8-53" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-54"><a href="#cb8-54" tabindex="-1"></a><span class="co">#&gt; === [III/III] Genome-wide CACTI peak-window pipeline completed with FDR! ===</span></span>
<span id="cb8-55"><a href="#cb8-55" tabindex="-1"></a><span class="co">#&gt; Run summary:</span></span>
<span id="cb8-56"><a href="#cb8-56" tabindex="-1"></a><span class="co">#&gt; Pvalue output: /tmp/RtmpgYDXR8/cacti_genome_3137f182f5581_pval_window50kb_chr5.txt.gz</span></span>
<span id="cb8-57"><a href="#cb8-57" tabindex="-1"></a><span class="co">#&gt; FDR output: /tmp/RtmpgYDXR8/cacti_fdr_chr5.txt.gz</span></span>
<span id="cb8-58"><a href="#cb8-58" tabindex="-1"></a></span>
<span id="cb8-59"><a href="#cb8-59" tabindex="-1"></a>res_genome</span>
<span id="cb8-60"><a href="#cb8-60" tabindex="-1"></a><span class="co">#&gt; $file_peak_group</span></span>
<span id="cb8-61"><a href="#cb8-61" tabindex="-1"></a><span class="co">#&gt; [1] &quot;/tmp/RtmpgYDXR8/cacti_genome_3137f182f5581_peak_group_window50kb.txt&quot;</span></span>
<span id="cb8-62"><a href="#cb8-62" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-63"><a href="#cb8-63" tabindex="-1"></a><span class="co">#&gt; $file_peak_group_peaklevel</span></span>
<span id="cb8-64"><a href="#cb8-64" tabindex="-1"></a><span class="co">#&gt; [1] &quot;/tmp/RtmpgYDXR8/cacti_genome_3137f182f5581_peak_group_window50kb_peak_as_row.txt&quot;</span></span>
<span id="cb8-65"><a href="#cb8-65" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-66"><a href="#cb8-66" tabindex="-1"></a><span class="co">#&gt; $file_pheno_cov_residual</span></span>
<span id="cb8-67"><a href="#cb8-67" tabindex="-1"></a><span class="co">#&gt; [1] &quot;/tmp/RtmpgYDXR8/cacti_genome_3137f182f5581_pheno_cov_residual.txt&quot;</span></span>
<span id="cb8-68"><a href="#cb8-68" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-69"><a href="#cb8-69" tabindex="-1"></a><span class="co">#&gt; $file_p_peak_group</span></span>
<span id="cb8-70"><a href="#cb8-70" tabindex="-1"></a><span class="co">#&gt; [1] &quot;/tmp/RtmpgYDXR8/cacti_genome_3137f182f5581_pval_window50kb_chr5.txt.gz&quot;</span></span>
<span id="cb8-71"><a href="#cb8-71" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb8-72"><a href="#cb8-72" tabindex="-1"></a><span class="co">#&gt; $file_fdr_out</span></span>
<span id="cb8-73"><a href="#cb8-73" tabindex="-1"></a><span class="co">#&gt; [1] &quot;/tmp/RtmpgYDXR8/cacti_fdr_chr5.txt.gz&quot;</span></span></code></pre></div>
<hr />
</div>
<div id="summary" class="section level2">
<h2>7. Summary</h2>
<p>In this vignette, we:</p>
<ul>
<li>described the <strong>CACTI peak-window pipeline</strong>,<br />
</li>
<li>documented the expected <strong>input file formats</strong> for peak
metadata, phenotypes, covariates, and QTL summary statistics,</li>
<li>explained the main <strong>output files</strong> produced by
<code>cacti_run_chr()</code>, and<br />
</li>
<li>demonstrated how to run the pipeline per-chromosome and genome-wide
using bundled toy data in <code>inst/extdata/</code>.</li>
</ul>
<p>For more details on each step, see the function reference:</p>
<ul>
<li><code>?cacti_group_peak_window</code><br />
</li>
<li><code>?cacti_pheno_cov_residual</code><br />
</li>
<li><code>?cacti_cal_p</code><br />
</li>
<li><code>?cacti_add_fdr</code><br />
</li>
<li><code>?cacti_run_chr</code><br />
</li>
<li><code>?cacti_run_genome</code></li>
</ul>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
