<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>CACTI peak-window pipeline • cacti</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="CACTI peak-window pipeline">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">cacti</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../index.html">Home</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item"><a class="nav-link" href="../articles/index.html">Vignettes</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/liliw-w/cacti/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>CACTI peak-window pipeline</h1>
                        <h4 data-toc-skip class="author">Lili Wang</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/liliw-w/cacti/blob/HEAD/vignettes/cacti_peak_window.Rmd" class="external-link"><code>vignettes/cacti_peak_window.Rmd</code></a></small>
      <div class="d-none name"><code>cacti_peak_window.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">1. Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>The <code>cacti</code> package implements the <strong>CACTI
peak-window method</strong> for chromatin QTL mapping.<br>
It is designed to:</p>
<ul>
<li>group nearby peaks into <strong>non-overlapping
windows</strong>,<br>
</li>
<li>residualize peak intensities with respect to covariates,<br>
</li>
<li>test association between genetic variants and <strong>multi-peak
windows</strong> using a principal-component omnibus (PCO) test,
and<br>
</li>
<li>optionally aggregate per-chromosome results and compute
<strong>window-level FDR</strong>.</li>
</ul>
<p>The main user-facing functions in this vignette are:</p>
<ul>
<li>
<code><a href="../reference/cacti_run_chr.html">cacti_run_chr()</a></code> – run the full CACTI peak-window
pipeline for <strong>one chromosome</strong>,<br>
</li>
<li>
<code><a href="../reference/cacti_run_genome.html">cacti_run_genome()</a></code> – run the pipeline across
<strong>multiple chromosomes</strong> and add FDR.</li>
</ul>
<p>This vignette explains:</p>
<ol style="list-style-type: decimal">
<li>The overall workflow.<br>
</li>
<li>The expected <strong>input file formats</strong>.<br>
</li>
<li>The <strong>output files</strong> produced by the pipeline.<br>
</li>
<li>Example usage with bundled toy data under
<code>inst/extdata/</code>.</li>
</ol>
<hr>
</div>
<div class="section level2">
<h2 id="overview-of-the-peak-window-pipeline">2. Overview of the peak-window pipeline<a class="anchor" aria-label="anchor" href="#overview-of-the-peak-window-pipeline"></a>
</h2>
<p>The CACTI peak-window pipeline has three main steps:</p>
<ol style="list-style-type: decimal">
<li>
<strong>Group peaks into non-overlapping windows</strong>
<ul>
<li>Using genomic coordinates (chromosome, start, end) of peaks,<br>
</li>
<li>Divide the genome into fixed-size windows (e.g. 50 kb),<br>
</li>
<li>Assign peaks to these windows and create a <strong>window (group)
table</strong>.</li>
</ul>
</li>
<li>
<strong>Residualize phenotypes with respect to covariates</strong>
<ul>
<li>Input: peak intensity matrix (features x samples) and covariate
matrix (covariates x samples),<br>
</li>
<li>Fit linear models <code>peak ~ covariates</code>,<br>
</li>
<li>Save residualized intensity values in the <strong>same
format</strong> as the original matrix, as one of the inputs for the
association test.</li>
</ul>
</li>
<li>
<strong>Run association tests on peak windows</strong>
<ul>
<li>Input: cis-QTL summary statistics <code>z</code> for peak–SNP pairs,
peak windows, and residualized phenotype,<br>
</li>
<li>For each window:
<ul>
<li>construct a Z-score matrix (variants x peaks),</li>
<li>compute a PCO-based multi-peak p-value if the number of peaks ≥
<code>min_peaks</code>,</li>
<li>otherwise fall back to a univariate p-value,</li>
</ul>
</li>
<li>Output: one p-value file per chromosome, one row per
<code>(group, snp)</code> pair.</li>
</ul>
</li>
</ol>
<p>Across all chromosomes, <code><a href="../reference/cacti_add_fdr.html">cacti_add_fdr()</a></code> (used by
<code><a href="../reference/cacti_run_genome.html">cacti_run_genome()</a></code>) aggregates p-values and computes
<strong>q-values</strong> for the top hit per window.</p>
<hr>
</div>
<div class="section level2">
<h2 id="input-file-formats">3. Input file formats<a class="anchor" aria-label="anchor" href="#input-file-formats"></a>
</h2>
<p>This section describes the expected columns and layout of each input
file type.</p>
<p>We will also inspect the <strong>toy dataset</strong> bundled with
the package in:</p>
<pre class="text"><code><span><span class="va">inst</span><span class="op">/</span><span class="va">extdata</span><span class="op">/</span></span>
<span>  <span class="va">test_pheno_meta.bed</span></span>
<span>  <span class="va">test_pheno.txt</span></span>
<span>  <span class="va">test_covariates.txt</span></span>
<span>  <span class="va">test_qtl_sum_stats_chr5.txt.gz</span></span></code></pre>
<div class="section level3">
<h3 id="peak-metadata-pheno-meta-bed-like">3.1 Peak metadata (pheno meta / BED-like)<a class="anchor" aria-label="anchor" href="#peak-metadata-pheno-meta-bed-like"></a>
</h3>
<p>The <strong>peak metadata</strong> file describes the genomic
coordinates and IDs of peaks.</p>
<ul>
<li>Used by: <code><a href="../reference/cacti_group_peak_window.html">cacti_group_peak_window()</a></code>,
<code><a href="../reference/cacti_run_chr.html">cacti_run_chr()</a></code>, <code><a href="../reference/cacti_run_genome.html">cacti_run_genome()</a></code>
</li>
<li>Argument: <code>file_pheno_meta</code>
</li>
<li>Example file: <code>test_pheno_meta.bed</code>
</li>
</ul>
<p>Required columns:</p>
<ul>
<li>
<code>phe_chr</code> – chromosome name, typically with
<code>"chr"</code> prefix (e.g. <code>"chr5"</code>),<br>
</li>
<li>
<code>phe_from</code> – peak start position (integer; must be
<code>&lt; phe_to</code>),<br>
</li>
<li>
<code>phe_to</code> – peak end position (integer),<br>
</li>
<li>
<code>phe_id</code> – unique peak identifier.</li>
</ul>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/liliw-w/cacti" class="external-link">cacti</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">file_pheno_meta</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span></span>
<span>  <span class="st">"extdata"</span>, <span class="st">"test_pheno_meta.bed"</span>,</span>
<span>  package <span class="op">=</span> <span class="st">"cacti"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu">data.table</span><span class="fu">::</span><span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/fread.html" class="external-link">fread</a></span><span class="op">(</span><span class="va">file_pheno_meta</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;                phe_id phe_chr phe_from phe_to</span></span>
<span><span class="co">#&gt;                &lt;char&gt;  &lt;char&gt;    &lt;int&gt;  &lt;int&gt;</span></span>
<span><span class="co">#&gt; 1:   chr5_11429_11873    chr5    11429  11873</span></span>
<span><span class="co">#&gt; 2: chr5_217285_220962    chr5   217285 220962</span></span>
<span><span class="co">#&gt; 3: chr5_223012_223844    chr5   223012 223844</span></span>
<span><span class="co">#&gt; 4: chr5_270610_273325    chr5   270610 273325</span></span>
<span><span class="co">#&gt; 5: chr5_320783_323201    chr5   320783 323201</span></span>
<span><span class="co">#&gt; 6: chr5_343419_345060    chr5   343419 345060</span></span></code></pre></div>
<hr>
</div>
<div class="section level3">
<h3 id="phenotype-matrix-peak-intensity">3.2 Phenotype matrix (peak intensity)<a class="anchor" aria-label="anchor" href="#phenotype-matrix-peak-intensity"></a>
</h3>
<p>The <strong>phenotype matrix</strong> contains peak intensities (or
other quantitative trait values).</p>
<ul>
<li>Used by: <code><a href="../reference/cacti_pheno_cov_residual.html">cacti_pheno_cov_residual()</a></code>,
<code><a href="../reference/cacti_run_chr.html">cacti_run_chr()</a></code>, <code><a href="../reference/cacti_run_genome.html">cacti_run_genome()</a></code>
</li>
<li>Argument: <code>file_pheno</code>
</li>
<li>Example file: <code>test_pheno.txt</code>
</li>
</ul>
<p>Layout:</p>
<ul>
<li>Rows = features (peaks).<br>
</li>
<li>First column = feature ID (matching <code>phe_id</code> in
metadata).<br>
</li>
<li>Remaining columns = samples.</li>
</ul>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">file_pheno</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span></span>
<span>  <span class="st">"extdata"</span>, <span class="st">"test_pheno.txt"</span>,</span>
<span>  package <span class="op">=</span> <span class="st">"cacti"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu">data.table</span><span class="fu">::</span><span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/fread.html" class="external-link">fread</a></span><span class="op">(</span><span class="va">file_pheno</span><span class="op">)</span><span class="op">)</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span>
<span><span class="co">#&gt;                        ID       AF04      AF06       AF08      AF12</span></span>
<span><span class="co">#&gt;                    &lt;char&gt;      &lt;num&gt;     &lt;num&gt;      &lt;num&gt;     &lt;num&gt;</span></span>
<span><span class="co">#&gt; 1: chr5_10248638_10251938  6.1639272 6.5171965  6.0122366 6.0046710</span></span>
<span><span class="co">#&gt; 2: chr5_10264481_10265888  0.2981699 0.2977740  2.0098907 0.4789852</span></span>
<span><span class="co">#&gt; 3: chr5_10265755_10268539  1.6214124 2.1306168  3.1048357 0.7830301</span></span>
<span><span class="co">#&gt; 4: chr5_10307340_10308285  0.7222144 0.8835924  0.2871122 0.5861384</span></span>
<span><span class="co">#&gt; 5: chr5_10311409_10312352  0.0350794 1.3397306 -0.2311209 0.3488682</span></span>
<span><span class="co">#&gt; 6: chr5_10333519_10334110 -0.1660226 0.8808080  0.5575939 1.1806155</span></span></code></pre></div>
<hr>
</div>
<div class="section level3">
<h3 id="covariate-matrix">3.3 Covariate matrix<a class="anchor" aria-label="anchor" href="#covariate-matrix"></a>
</h3>
<p>The <strong>covariate matrix</strong> encodes covariates for each
sample (e.g. PCs, batch, sex, etc.).</p>
<ul>
<li>Used by: <code><a href="../reference/cacti_pheno_cov_residual.html">cacti_pheno_cov_residual()</a></code>,
<code><a href="../reference/cacti_run_chr.html">cacti_run_chr()</a></code>, <code><a href="../reference/cacti_run_genome.html">cacti_run_genome()</a></code>
</li>
<li>Argument: <code>file_cov</code>
</li>
<li>Example file: <code>test_covariates.txt</code>
</li>
</ul>
<p>Layout:</p>
<ul>
<li>Rows = covariates.<br>
</li>
<li>First column = covariate ID.<br>
</li>
<li>Remaining columns = samples.</li>
</ul>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">file_cov</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span></span>
<span>  <span class="st">"extdata"</span>, <span class="st">"test_covariates.txt"</span>,</span>
<span>  package <span class="op">=</span> <span class="st">"cacti"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu">data.table</span><span class="fu">::</span><span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/fread.html" class="external-link">fread</a></span><span class="op">(</span><span class="va">file_cov</span><span class="op">)</span><span class="op">)</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span>
<span><span class="co">#&gt;        ID       AF04       AF06       AF08       AF12</span></span>
<span><span class="co">#&gt;    &lt;char&gt;      &lt;num&gt;      &lt;num&gt;      &lt;num&gt;      &lt;num&gt;</span></span>
<span><span class="co">#&gt; 1:   cov1 -0.1696051 -0.2304933 -0.2132352 -0.2220646</span></span></code></pre></div>
<hr>
</div>
<div class="section level3">
<h3 id="qtl-summary-statistics">3.4 QTL summary statistics<a class="anchor" aria-label="anchor" href="#qtl-summary-statistics"></a>
</h3>
<p>The <strong>cis-QTL summary stats</strong> file contains association
summary statistics for peak–SNP pairs.</p>
<ul>
<li>Used by: <code><a href="../reference/cacti_cal_p.html">cacti_cal_p()</a></code>, <code><a href="../reference/cacti_run_chr.html">cacti_run_chr()</a></code>,
<code><a href="../reference/cacti_run_genome.html">cacti_run_genome()</a></code>
</li>
<li>Argument: <code>qtl_file</code> (per chromosome) or
<code>qtl_files</code> (multiple chromosomes, given as a
vector/template)</li>
<li>Example file: <code>test_qtl_sum_stats_chr5.txt.gz</code>
</li>
</ul>
<p>Required columns:</p>
<ul>
<li>
<code>phe_id</code> – peak ID (matching <code>phe_id</code> in
metadata and <code>ID</code> in phenotype matrix),<br>
</li>
<li>
<code>var_id</code> – variant ID,<br>
</li>
<li>
<code>z</code> – z-score for the association between that variant
and that peak.</li>
</ul>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">qtl_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span></span>
<span>  <span class="st">"extdata"</span>, <span class="st">"test_qtl_sum_stats_chr5.txt.gz"</span>,</span>
<span>  package <span class="op">=</span> <span class="st">"cacti"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu">data.table</span><span class="fu">::</span><span class="fu"><a href="https://rdatatable.gitlab.io/data.table/reference/fread.html" class="external-link">fread</a></span><span class="op">(</span><span class="va">qtl_file</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;              phe_id      var_id         z</span></span>
<span><span class="co">#&gt;              &lt;char&gt;      &lt;char&gt;     &lt;num&gt;</span></span>
<span><span class="co">#&gt; 1: chr5_11429_11873 5_12041_T_A 0.2715305</span></span>
<span><span class="co">#&gt; 2: chr5_11429_11873 5_12114_G_C 0.1525914</span></span>
<span><span class="co">#&gt; 3: chr5_11429_11873 5_12225_A_G 1.0831622</span></span>
<span><span class="co">#&gt; 4: chr5_11429_11873 5_13018_A_C 0.9401877</span></span>
<span><span class="co">#&gt; 5: chr5_11429_11873 5_13114_C_G 0.9401877</span></span>
<span><span class="co">#&gt; 6: chr5_11429_11873 5_13125_T_G 0.9401877</span></span></code></pre></div>
<hr>
</div>
</div>
<div class="section level2">
<h2 id="output-files-from-cacti_run_chr">4. Output files from <code>cacti_run_chr()</code><a class="anchor" aria-label="anchor" href="#output-files-from-cacti_run_chr"></a>
</h2>
<p><code><a href="../reference/cacti_run_chr.html">cacti_run_chr()</a></code> is the main function for running the
CACTI peak-window pipeline on a <strong>single chromosome</strong>.</p>
<div class="section level3">
<h3 id="overview">4.1 Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h3>
<p>Calling:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a>res_single_chr <span class="ot">&lt;-</span> <span class="fu">cacti_run_chr</span>(</span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a>  <span class="at">window_size     =</span> <span class="st">"50kb"</span>,</span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a>  <span class="at">file_pheno_meta =</span> <span class="sc">&lt;</span>peak metadata<span class="sc">&gt;</span>,</span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a>  <span class="at">file_pheno      =</span> <span class="sc">&lt;</span>phenotype matrix<span class="sc">&gt;</span>,</span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a>  <span class="at">file_cov        =</span> <span class="sc">&lt;</span>covariate matrix<span class="sc">&gt;</span>,</span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a>  <span class="at">chr             =</span> <span class="st">"chr5"</span>,</span>
<span id="cb6-7"><a href="#cb6-7" tabindex="-1"></a>  <span class="at">qtl_file        =</span> <span class="sc">&lt;</span>QTL summary stats <span class="cf">for</span> chr5<span class="sc">&gt;</span>,</span>
<span id="cb6-8"><a href="#cb6-8" tabindex="-1"></a>  <span class="at">out_prefix      =</span> <span class="sc">&lt;</span>output prefix<span class="sc">&gt;</span>,</span>
<span id="cb6-9"><a href="#cb6-9" tabindex="-1"></a>  <span class="at">min_peaks       =</span> <span class="dv">2</span></span>
<span id="cb6-10"><a href="#cb6-10" tabindex="-1"></a>)</span></code></pre></div>
<p>will:</p>
<ol style="list-style-type: decimal">
<li>Construct peak windows on all chromosomes (using
<code>window_size</code>).<br>
</li>
<li>Residualize the phenotype matrix.<br>
</li>
<li>Run the CACTI association test on the specified chromosome
(<code>chr</code>).</li>
</ol>
<p>It returns a named list of output paths, and writes the following
files:</p>
</div>
<div class="section level3">
<h3 id="window-level-grouping-file">4.2 Window-level grouping file<a class="anchor" aria-label="anchor" href="#window-level-grouping-file"></a>
</h3>
<ul>
<li>Path: <code>res_single_chr$file_peak_group</code><br>
</li>
<li>Produced by: <code><a href="../reference/cacti_group_peak_window.html">cacti_group_peak_window()</a></code>
</li>
</ul>
<p>Columns:</p>
<ul>
<li>
<code>group</code> – window (group) ID (e.g. <code>"G1"</code>,
<code>"G2"</code>, …),<br>
</li>
<li>
<code>group_chr</code> – chromosome for the window,<br>
</li>
<li>
<code>group_from</code> – window start position,<br>
</li>
<li>
<code>group_to</code> – window end position,<br>
</li>
<li>
<code>pid_group</code> – semicolon-separated list of peak IDs in the
window,<br>
</li>
<li>
<code>n_pid_group</code> – number of peaks in the window.</li>
</ul>
</div>
<div class="section level3">
<h3 id="peak-level-grouping-file">4.3 Peak-level grouping file<a class="anchor" aria-label="anchor" href="#peak-level-grouping-file"></a>
</h3>
<ul>
<li>Path: <code>res_single_chr$file_peak_group_peaklevel</code>
</li>
</ul>
<p>Columns:</p>
<ul>
<li>
<code>phe_id</code> – peak ID,<br>
</li>
<li>
<code>phe_chr</code> – chromosome,<br>
</li>
<li>
<code>phe_from</code>, <code>phe_to</code> – peak start/end,<br>
</li>
<li>
<code>group</code> – window ID to which the peak belongs,<br>
</li>
<li>
<code>n_pid_group</code> – number of peaks in that window.</li>
</ul>
</div>
<div class="section level3">
<h3 id="residualized-phenotype-matrix">4.4 Residualized phenotype matrix<a class="anchor" aria-label="anchor" href="#residualized-phenotype-matrix"></a>
</h3>
<ul>
<li>Path: <code>res_single_chr$file_pheno_cov_residual</code>
</li>
</ul>
<p>Same layout as the input phenotype matrix:</p>
<ul>
<li>Rows = features (peak IDs),<br>
</li>
<li>First column = <code>ID</code> (peak IDs),<br>
</li>
<li>Remaining columns = residualized values for each sample.</li>
</ul>
</div>
<div class="section level3">
<h3 id="per-window-p-value-file-for-the-chromosome">4.5 Per-window p-value file for the chromosome<a class="anchor" aria-label="anchor" href="#per-window-p-value-file-for-the-chromosome"></a>
</h3>
<ul>
<li>Path: <code>res_single_chr$file_p_peak_group</code>
</li>
</ul>
<p>Each row corresponds to a <code>(group, snp)</code> pair:</p>
<ul>
<li>
<code>group</code> – window ID,<br>
</li>
<li>
<code>snp</code> – variant ID,<br>
</li>
<li>
<code>pval</code> – window-level p-value for that variant.
<ul>
<li>For windows with ≥ <code>min_peaks</code> peaks, this is the
PCO-based multi-peak p-value.<br>
</li>
<li>For single-peak windows, it is a univariate p-value derived from
<code>z</code>.</li>
</ul>
</li>
</ul>
<hr>
</div>
</div>
<div class="section level2">
<h2 id="per-chromosome-example-cacti_run_chr">5. Per-chromosome example: <code>cacti_run_chr()</code><a class="anchor" aria-label="anchor" href="#per-chromosome-example-cacti_run_chr"></a>
</h2>
<p>Below is an example of running the full pipeline on the <strong>toy
chr5 data</strong> included in <code>inst/extdata/</code>.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/liliw-w/cacti" class="external-link">cacti</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">file_pheno_meta</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span></span>
<span>  <span class="st">"extdata"</span>, <span class="st">"test_pheno_meta.bed"</span>,</span>
<span>  package <span class="op">=</span> <span class="st">"cacti"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">file_pheno</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span></span>
<span>  <span class="st">"extdata"</span>, <span class="st">"test_pheno.txt"</span>,</span>
<span>  package <span class="op">=</span> <span class="st">"cacti"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">file_cov</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span></span>
<span>  <span class="st">"extdata"</span>, <span class="st">"test_covariates.txt"</span>,</span>
<span>  package <span class="op">=</span> <span class="st">"cacti"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">qtl_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span></span>
<span>  <span class="st">"extdata"</span>, <span class="st">"test_qtl_sum_stats_chr5.txt.gz"</span>,</span>
<span>  package <span class="op">=</span> <span class="st">"cacti"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Use a temporary prefix to avoid cluttering the working directory</span></span>
<span><span class="va">out_prefix</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="st">"cacti_chr5_"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">res_single_chr</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cacti_run_chr.html">cacti_run_chr</a></span><span class="op">(</span></span>
<span>  window_size     <span class="op">=</span> <span class="st">"50kb"</span>,</span>
<span>  file_pheno_meta <span class="op">=</span> <span class="va">file_pheno_meta</span>,</span>
<span>  file_pheno      <span class="op">=</span> <span class="va">file_pheno</span>,</span>
<span>  file_cov        <span class="op">=</span> <span class="va">file_cov</span>,</span>
<span>  chr             <span class="op">=</span> <span class="st">"chr5"</span>,</span>
<span>  qtl_file        <span class="op">=</span> <span class="va">qtl_file</span>,</span>
<span>  out_prefix      <span class="op">=</span> <span class="va">out_prefix</span>,</span>
<span>  min_peaks       <span class="op">=</span> <span class="fl">2</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; [1/3] Group peaks into non-overlapping windows …</span></span>
<span><span class="co">#&gt; [2/3] Residualize phenotype by covariates …</span></span>
<span><span class="co">#&gt; [3/3] Run pvalue for chr5 …</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  CACTI peak-window pipeline for chr5 completed successfully!</span></span>
<span><span class="co">#&gt; Run summary:</span></span>
<span><span class="co">#&gt; Chromosome:chr5</span></span>
<span><span class="co">#&gt; Window size:50kb</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Input files:</span></span>
<span><span class="co">#&gt; Peaks BED:/tmp/RtmpIvhooj/temp_libpathfd7a43f04ed54/cacti/extdata/test_pheno_meta.bed</span></span>
<span><span class="co">#&gt; Phenotype:/tmp/RtmpIvhooj/temp_libpathfd7a43f04ed54/cacti/extdata/test_pheno.txt</span></span>
<span><span class="co">#&gt; Covariate:/tmp/RtmpIvhooj/temp_libpathfd7a43f04ed54/cacti/extdata/test_covariates.txt</span></span>
<span><span class="co">#&gt; QTL summary stats:/tmp/RtmpIvhooj/temp_libpathfd7a43f04ed54/cacti/extdata/test_qtl_sum_stats_chr5.txt.gz</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Output files:</span></span>
<span><span class="co">#&gt; Grouped peak: /tmp/Rtmpldbpiw/cacti_chr5_fe8494aa45928_peak_group_window50kb.txt</span></span>
<span><span class="co">#&gt; Grouped peak (peak as row):/tmp/Rtmpldbpiw/cacti_chr5_fe8494aa45928_peak_group_window50kb_peak_as_row.txt</span></span>
<span><span class="co">#&gt; Residual phenotype:/tmp/Rtmpldbpiw/cacti_chr5_fe8494aa45928_pheno_cov_residual.txt</span></span>
<span><span class="co">#&gt; Pval results (chr5): /tmp/Rtmpldbpiw/cacti_chr5_fe8494aa45928_pval_window50kb_chr5.txt.gz</span></span>
<span></span>
<span><span class="va">res_single_chr</span></span>
<span><span class="co">#&gt; $file_peak_group</span></span>
<span><span class="co">#&gt; [1] "/tmp/Rtmpldbpiw/cacti_chr5_fe8494aa45928_peak_group_window50kb.txt"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $file_peak_group_peaklevel</span></span>
<span><span class="co">#&gt; [1] "/tmp/Rtmpldbpiw/cacti_chr5_fe8494aa45928_peak_group_window50kb_peak_as_row.txt"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $file_pheno_cov_residual</span></span>
<span><span class="co">#&gt; [1] "/tmp/Rtmpldbpiw/cacti_chr5_fe8494aa45928_pheno_cov_residual.txt"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $file_p_peak_group</span></span>
<span><span class="co">#&gt; [1] "/tmp/Rtmpldbpiw/cacti_chr5_fe8494aa45928_pval_window50kb_chr5.txt.gz"</span></span></code></pre></div>
<hr>
</div>
<div class="section level2">
<h2 id="genome-wide-wrapper-cacti_run_genome">6. Genome-wide wrapper: <code>cacti_run_genome()</code><a class="anchor" aria-label="anchor" href="#genome-wide-wrapper-cacti_run_genome"></a>
</h2>
<p><code><a href="../reference/cacti_run_genome.html">cacti_run_genome()</a></code> provides a convenience wrapper for
running the pipeline across <strong>multiple chromosomes</strong>, then
aggregating the p-values and adding FDR via
<code><a href="../reference/cacti_add_fdr.html">cacti_add_fdr()</a></code>.</p>
<p>Conceptually, it:</p>
<ol style="list-style-type: decimal">
<li>Loops over <code>chrs</code> and calls <code><a href="../reference/cacti_run_chr.html">cacti_run_chr()</a></code>
for each one.<br>
</li>
<li>Collects all per-chromosome p-value files.<br>
</li>
<li>Calls <code><a href="../reference/cacti_add_fdr.html">cacti_add_fdr()</a></code> once to compute window-level
q-values across all windows and chromosomes.</li>
</ol>
<p>The simplest usage is when you have one QTL file per chromosome and
pass either:</p>
<ul>
<li>a character vector <code>qtl_files</code> with one path per
chromosome, or<br>
</li>
<li>a template with <code>"{chr}"</code> placeholder,
e.g. <code>"extdata/test_qtl_sum_stats_{chr}.txt.gz"</code>.</li>
</ul>
<p>In the toy example, we only use data for <strong>chr5</strong>, but
this still demonstrates how to call the function.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">file_pheno_meta</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span></span>
<span>  <span class="st">"extdata"</span>, <span class="st">"test_pheno_meta.bed"</span>,</span>
<span>  package <span class="op">=</span> <span class="st">"cacti"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">file_pheno</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span></span>
<span>  <span class="st">"extdata"</span>, <span class="st">"test_pheno.txt"</span>,</span>
<span>  package <span class="op">=</span> <span class="st">"cacti"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">file_cov</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span></span>
<span>  <span class="st">"extdata"</span>, <span class="st">"test_covariates.txt"</span>,</span>
<span>  package <span class="op">=</span> <span class="st">"cacti"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">qtl_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span></span>
<span>  <span class="st">"extdata"</span>, <span class="st">"test_qtl_sum_stats_chr5.txt.gz"</span>,</span>
<span>  package <span class="op">=</span> <span class="st">"cacti"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">out_prefix</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="st">"cacti_genome_"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">res_genome</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/cacti_run_genome.html">cacti_run_genome</a></span><span class="op">(</span></span>
<span>  window_size     <span class="op">=</span> <span class="st">"50kb"</span>,</span>
<span>  file_pheno_meta <span class="op">=</span> <span class="va">file_pheno_meta</span>,</span>
<span>  file_pheno      <span class="op">=</span> <span class="va">file_pheno</span>,</span>
<span>  file_cov        <span class="op">=</span> <span class="va">file_cov</span>,</span>
<span>  chrs            <span class="op">=</span> <span class="st">"chr5"</span>,                 <span class="co"># toy example: one chromosome</span></span>
<span>  qtl_files       <span class="op">=</span> <span class="va">qtl_file</span>,</span>
<span>  out_prefix      <span class="op">=</span> <span class="va">out_prefix</span>,</span>
<span>  file_fdr_out    <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>, <span class="st">"cacti_fdr_chr5.txt.gz"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; === [I/III] Running CACTI peak-window pipeline genome-wide ===</span></span>
<span><span class="co">#&gt; [1/3] Group peaks into non-overlapping windows …</span></span>
<span><span class="co">#&gt; [2/3] Residualize phenotype by covariates …</span></span>
<span><span class="co">#&gt; [3/3] Run pvalue for chr5 …</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  CACTI peak-window pipeline for chr5 completed successfully!</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Input files:</span></span>
<span><span class="co">#&gt; Peaks BED:/tmp/RtmpIvhooj/temp_libpathfd7a43f04ed54/cacti/extdata/test_pheno_meta.bed</span></span>
<span><span class="co">#&gt; Phenotype:/tmp/RtmpIvhooj/temp_libpathfd7a43f04ed54/cacti/extdata/test_pheno.txt</span></span>
<span><span class="co">#&gt; Covariate:/tmp/RtmpIvhooj/temp_libpathfd7a43f04ed54/cacti/extdata/test_covariates.txt</span></span>
<span><span class="co">#&gt; QTL summary stats:/tmp/RtmpIvhooj/temp_libpathfd7a43f04ed54/cacti/extdata/test_qtl_sum_stats_chr5.txt.gz</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Output files:</span></span>
<span><span class="co">#&gt; Grouped peak: /tmp/Rtmpldbpiw/cacti_genome_fe8497e195629_peak_group_window50kb.txt</span></span>
<span><span class="co">#&gt; Grouped peak (peak as row):/tmp/Rtmpldbpiw/cacti_genome_fe8497e195629_peak_group_window50kb_peak_as_row.txt</span></span>
<span><span class="co">#&gt; Residual phenotype:/tmp/Rtmpldbpiw/cacti_genome_fe8497e195629_pheno_cov_residual.txt</span></span>
<span><span class="co">#&gt; Pval results (chr5): /tmp/Rtmpldbpiw/cacti_genome_fe8497e195629_pval_window50kb_chr5.txt.gz</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; === [II/III] Adding FDR across windows and chromosomes ===</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; === [III/III] Genome-wide CACTI peak-window pipeline completed with FDR! ===</span></span>
<span><span class="co">#&gt; Run summary:</span></span>
<span><span class="co">#&gt; Pvalue output: /tmp/Rtmpldbpiw/cacti_genome_fe8497e195629_pval_window50kb_chr5.txt.gz</span></span>
<span><span class="co">#&gt; FDR output: /tmp/Rtmpldbpiw/cacti_fdr_chr5.txt.gz</span></span>
<span></span>
<span><span class="va">res_genome</span></span>
<span><span class="co">#&gt; $file_peak_group</span></span>
<span><span class="co">#&gt; [1] "/tmp/Rtmpldbpiw/cacti_genome_fe8497e195629_peak_group_window50kb.txt"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $file_peak_group_peaklevel</span></span>
<span><span class="co">#&gt; [1] "/tmp/Rtmpldbpiw/cacti_genome_fe8497e195629_peak_group_window50kb_peak_as_row.txt"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $file_pheno_cov_residual</span></span>
<span><span class="co">#&gt; [1] "/tmp/Rtmpldbpiw/cacti_genome_fe8497e195629_pheno_cov_residual.txt"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $file_p_peak_group</span></span>
<span><span class="co">#&gt; [1] "/tmp/Rtmpldbpiw/cacti_genome_fe8497e195629_pval_window50kb_chr5.txt.gz"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $file_fdr_out</span></span>
<span><span class="co">#&gt; [1] "/tmp/Rtmpldbpiw/cacti_fdr_chr5.txt.gz"</span></span></code></pre></div>
<hr>
</div>
<div class="section level2">
<h2 id="summary">7. Summary<a class="anchor" aria-label="anchor" href="#summary"></a>
</h2>
<p>In this vignette, we:</p>
<ul>
<li>described the <strong>CACTI peak-window pipeline</strong>,<br>
</li>
<li>documented the expected <strong>input file formats</strong> for peak
metadata, phenotypes, covariates, and QTL summary statistics,</li>
<li>explained the main <strong>output files</strong> produced by
<code><a href="../reference/cacti_run_chr.html">cacti_run_chr()</a></code>, and<br>
</li>
<li>demonstrated how to run the pipeline per-chromosome and genome-wide
using bundled toy data in <code>inst/extdata/</code>.</li>
</ul>
<p>For more details on each step, see the function reference:</p>
<ul>
<li>
<code><a href="../reference/cacti_group_peak_window.html">?cacti_group_peak_window</a></code><br>
</li>
<li>
<code><a href="../reference/cacti_pheno_cov_residual.html">?cacti_pheno_cov_residual</a></code><br>
</li>
<li>
<code><a href="../reference/cacti_cal_p.html">?cacti_cal_p</a></code><br>
</li>
<li>
<code><a href="../reference/cacti_add_fdr.html">?cacti_add_fdr</a></code><br>
</li>
<li>
<code><a href="../reference/cacti_run_chr.html">?cacti_run_chr</a></code><br>
</li>
<li><code><a href="../reference/cacti_run_genome.html">?cacti_run_genome</a></code></li>
</ul>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Lili Wang.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
